client/state/utils/test/integration.js

```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import { spy } from 'sinon';
+/**
+ * Internal dependencies
+ */
+import { uniqueId as uniqueId } from '../actions';
+import { uniqueId as uniqueIdReducer } from '../reducer';
+
+describe( 'integrationTest', () => {
+	describe( 'uniqueId', () => {
+		let state = 0;
+		const getState = () => {
+			return {
+				utils: {
+					uniqueId: state
+				}
+			};
+		};
+		const dispatchSpy = spy( action => {
+			state = uniqueIdReducer( state, action );
+		} );
+		const getStateSpy = spy( getState );
+
+		it( 'should return incremented number', () => {
+			const initState = 1;
+			state = initState;
+
+			const returnValue = uniqueId()( dispatchSpy, getStateSpy );
+
+			expect( state ).to.eql( initState + 1 );
+			expect( returnValue ).to.eql( state.toString() );
+		} );
+	} );
+} );
```

Comment:
I love the test in `test/reducer.js` and it appears to cover the behavior of the unique id. This "integration" test also does its job but there's no real distinction between it and the other unit tests because it runs every time with them. It's also not particularly slower then them anyway.

So since we know from the unit test that the `uniqueId()` function works as expected, I don't think we need this test too. It's mainly actually testing that redux does its job, which is covered by numerous redux unit tests.